src/coreclr/gc/gcpriv.h

```diff
@@ -4299,142 +4489,472 @@ class gc_heap
             return copied_count;
         }
 
-        int highest_avg_recorded_tcp (int count, float avg, float* highest_avg)
+        void init_recorded_tcp ()
+        {
+            total_recorded_tcp = 0;
+            recorded_tcp_index = 0;
+            dprintf (6666, ("INIT tcp buffer"));
+        }
+
+        int get_recorded_tcp_count () { return total_recorded_tcp; }
+
+        float           around_target_accumulation;
+        float           around_target_threshold;
+
+        bool is_tcp_in_range (float diff_pct, float slope)
+        {
+            return ((diff_pct <= 0.2) && (diff_pct >= -0.2) && (slope <= 0.1) && (slope >= -0.1));
+        }
+
+        bool is_close_to_max (int new_n, int max)
+        {
+            return ((max - new_n) <= (max / 10));
+        }
+
+        float slope (float* y, int n, float* avg);
+
+        // if the last attempt was successful, and we still aren't to target, we should be more aggressive.
+        int get_aggressiveness (int change_int)
         {
-            float highest_sum = 0.0;
-            int highest_count = 0;
+            int factor = 1;
 
-            for (int i = 0; i < count; i++)
+            adjust_metric metric = adjust_hc;
+
+            // Looking at the last 2 adjustments was too aggressive - so currently only look at the last one.
+            for (int i = -1; i >= -1; i--)
```

Comment:
is this loop necessary?

Reply:
as the comment says this used to be 2 times (that's why I had a loop). I left the loop when I changed it to 1 time because I thought it might be useful later if we decided to look at more entries (but with some other policy).